### ISSUE TITLE
SDK Cognito Credentials Provider

### ISSUE THREAD

**Issue by dayaffe** [awslabs/aws-sdk-swift#1082]
### Describe the feature

Swift SDK should expose a credentials provider for [Cognito](https://aws.amazon.com/cognito/).

### Use Case

AWS Cognito is a user identity and access management service used by web and mobile applications. Since Swift is heavily used by mobile development, this feature would likely be used often.

### Proposed Solution

CRT needs to complete their analysis and complete any associated dev work before we can proceed with this change (See [CRT #189](https://github.com/awslabs/aws-crt-swift/issues/189)).

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

**Comment by dayaffe** [comment]
Reached out to @issmith from the Kotlin SDK team on whether they plan to support Cognito Credentials Provider as a language heavily leveraged by mobile applications (Android). They mentioned:

```
We're not planning to build a Cognito credentials provider. There's no SEP for it, customers aren't asking for it yet, and in the absence of those two things we can't be sure we'd build the right thing.

If a SEP or customer request appears, we'd reevaluate.
```

For now this feature may be put on hold.

**Comment by mikemike396** [comment]
We would like to see a Cognito credentials provider added. We're migrating from the [legacy sdk](https://github.com/aws-amplify/aws-sdk-ios) and have no option but to move to static credentials.

**Comment by dayaffe** [comment]
Great news, a cognito credentials provider is on the way! Will update here when it is released.
